packages/apollo/src/commands/service/check.ts

```diff
@@ -253,112 +260,198 @@ export default class ServiceCheck extends ProjectCommand {
     const breakingChangesErrorMessage = "breaking changes found";
-    const compositionErrorMessage = "composition errors found";
 
-    let schema, schemaHash;
+    let schema: GraphQLSchema | undefined;
     try {
       await this.runTasks<TasksOutput>(
         ({ config, flags, project }) => {
+          if (!isServiceProject(project)) {
+            throw new Error("`project` must be a `GraphQLServiceProject`");
+          }
+
           const configName = config.name;
           const tag = flags.tag || config.tag || "current";
+          const serviceName: string | undefined = flags.serviceName;
 
           if (!configName) {
             throw new Error("No service found to link to Engine");
           }
 
           return [
             {
-              title: `Validating local schema against tag ${chalk.blue(
-                tag
-              )} on service ${chalk.blue(configName)}`,
+              enabled: () => !!serviceName,
+              title: `Validate graph composition for service ${chalk.blue(
+                serviceName || ""
+              )} on graph ${chalk.blue(configName)}`,
               task: async (ctx: TasksOutput, task) => {
-                task.output = "Resolving schema";
-                taskOutput.shouldOutputJson = flags.json;
+                if (!serviceName) {
+                  throw new Error(
+                    "This task should not be run without a `serviceName`. Check the `enabled` function."
+                  );
+                }
+                task.output = "Fetching local service's partial schema";
 
-                if (flags.federated) {
-                  const info = await (project as GraphQLServiceProject).resolveFederationInfo();
-                  if (!info.sdl)
-                    throw new Error("No SDL found for federated service");
-
-                  /**
-                   * id: service id for root mutation (graph id)
-                   * variant: like a tag. prod/staging/etc
-                   * name: implementing service name inside of the graph
-                   * sha: git commit hash/docker id. placeholder for now
-                   */
-                  task.output = "Creating composed schema against the graph";
-                  const {
-                    errors,
-                    warnings,
-                    compositionValidationDetails
-                  } = await project.engine.checkPartialSchema({
-                    id: configName,
-                    graphVariant: tag,
-                    implementingServiceName: flags.serviceName || info.name,
-                    partialSchema: {
-                      sdl: info.sdl
+                const info = await project.resolveFederationInfo();
+                if (!info.sdl) {
+                  throw new Error("No SDL found for federated service");
+                }
+
+                task.output = `Attempting to compose graph with ${chalk.blue(
```

Comment:
It would be great to have some diagnostic information about how many other services are included in the composition

Reply:
Unloaded question: why do you think this is important @evans ?

Reply:
Good question! I think it's a nice to have, which would indicate how much work apollo is doing and a nice way for users to understand that the command is working as intended